pootle/apps/pootle_app/management/commands/calculate_checks.py

```diff
@@ -14,69 +14,99 @@
+from pootle_store.util import OBSOLETE
 
-logger = logging.getLogger('stats')
-cache = get_cache('stats')
+from . import PootleCommand
 
 
-class Command(RefreshStatsCommand):
-    help = "Allow stats and text indices to be refreshed manually."
+class Command(PootleCommand):
+    help = "Allow checks to be recalculated manually."
 
     shared_option_list = (
         make_option('--check', action='append', dest='check_names',
                     help='Check to recalculate'),
     )
-    cached_methods = [CachedMethods.CHECKS]
 
-    def handle_noargs(self, **options):
-        calculate_checks.delay(**options)
-        option_list = map(lambda x: '%s=%s' % (x, options[x]),
-                          filter(lambda x: options[x], options))
-        self.stdout.write('calculate checks RQ job added with options: %s. %s.' %
-                          (', '.join(option_list),
-                          'Please make sure rqworker is running'))
-
-    def process(self, **options):
-        check_names = options.get('check_names', [])
-        store_filter = options.get('store_filter', {})
-        unit_fk_filter = options.get('unit_fk_filter', {})
-        store_fk_filter = options.get('store_fk_filter', {})
-
-        logger.info('Initializing stores...')
-
-        stores = Store.objects.live()
-        if store_filter:
-            stores = stores.filter(**store_filter)
-
-        self._init_stores(stores)
-        self._init_checks()
-        self.calculate_checks(check_names, unit_fk_filter, store_fk_filter)
-
-        logger.info('Setting quality check stats values for all stores...')
-        self._set_qualitycheck_stats(unit_fk_filter)
-        logger.info('Setting empty values for other cache entries...')
-        self._set_empty_values()
-
-
-@job('default', timeout=18000)
-def calculate_checks(**options):
-    # The script prefix needs to be set here because the generated
-    # URLs need to be aware of that and they are cached. Ideally
-    # Django should take care of setting this up, but it doesn't yet:
-    # https://code.djangoproject.com/ticket/16734
-    script_name = (u'/' if settings.FORCE_SCRIPT_NAME is None
-                        else force_unicode(settings.FORCE_SCRIPT_NAME))
-    set_script_prefix(script_name)
-    super(RefreshStatsCommand, Command()).handle_noargs(**options)
+    cached_methods = [CachedMethods.CHECKS]
+    process_disabled_projects = True
+
+    def handle_all_stores(self, translation_project, **options):
+        self.process(translation_project=translation_project, **options)
+
+    def handle_all(self, **options):
+        if not self.projects and not self.languages:
+            logging.info(u"Running %s (noargs)", self.name)
+            try:
+                self.process(**options)
+            except Exception:
+                logging.exception(u"Failed to run %s", self.name)
+        else:
+            super(Command, self).handle_all(**options)
+
+    @staticmethod
```

Comment:
I don't understand why this has to be a static method. A simple function would work just fine.